File: String rp or pr.py
class Solution:
    def solve (self, X, Y, S):
        #code here
        answer=0
        a='pr'
        b='rp'
        if X<Y:
            X,Y=Y,X
            a,b=b,a
        st=[]
        for i in range(len(S)-1,-1,-1):
            if len(st)>0 and S[i]==a[0] and st[-1]==a[1]:
                st.pop(len(st)-1)
                answer+=X
            else:
                st.append(S[i])
        S=''
        while len(st)>0:
            S+=st[-1]
            st.pop(len(st)-1)
        for i in range(len(S)-1,-1,-1):
            if len(st)>0 and S[i]==b[0] and st[-1]==b[1]:
                st.pop(len(st)-1)
                answer+=Y
            else:
                st.append(S[i])
        return answer

File: String rp or pr.cpp
class Solution{   
public:

    long long solve(int X,int Y,string S){
      long long answer=0;
      string a="pr";
      string b="rp";
      if(X<Y)
      {
        swap(X,Y);
        swap(a,b);
      }

      stack<char> st;
      for(int i=(int)S.size()-1;i>=0;i--)
      {
        if(st.size()>0&&S[i]==a[0]&&st.top()==a[1])
        {
          st.pop();
          answer+=X;
        }
        else
        {
          st.push(S[i]);
        }
      }

      S="";
      while(st.size()>0){
        S.push_back(st.top());
        st.pop();
      }

      for(int i=(int)S.size()-1;i>=0;i--)
      {
        if(st.size()>0&&S[i]==b[0]&&st.top()==b[1])
        {
          st.pop();
          answer+=Y;
        }
        else
        {
          st.push(S[i]);
        }
      }
      return answer;
    }
};
